src/OnboardingSPA/pages/Steps/GetStarted/SiteTypeSetup/PrimarySite/index.js

```diff
@@ -94,63 +95,65 @@ const StepPrimarySetup = () => {
 						) }
 					/>
 				</div>
-
-				<div className="nfd-setup-primary-categories">
-					{ content.categories.map( ( item, idx ) => {
-						return (
-							<div
-								key={ item?.name }
-								className={ `${
-									clickedIndex === idx ||
-									item.name === selectedPrimaryCategoryInStore
-										? 'chosenPrimaryCategory '
-										: ''
-								}nfd-card-category` }
-								onClick={ ( e ) => handleCategoryClick( idx ) }
-							>
-								<div className="nfd-card-category-wrapper">
-									<span
-										className="icon"
-										style={ {
-											backgroundImage:
-												clickedIndex !== idx &&
-												item.name !==
-													selectedPrimaryCategoryInStore
-													? item?.icon
-													: item?.iconWhite,
-										} }
-									></span>
-									<span className="categName">
-										{ item?.name }
-									</span>
+				<StepStateHandler watch={ content.categories }>
```

Comment:
Let's also wait on `selectedPrimaryCategoryInStore`, the idea behind this story was that we don't flash in a selection after the categories load rather we fade in once we have highlighted the selection.

Reply:
good catch added to the check